test/tabs/test-firefox-tabs.js

```diff
@@ -1057,6 +1059,41 @@ exports.testFaviconGetterDeprecation = function (assert, done) {
   });
 }
 
+// related to bug 942511
+exports['test active tab properties defined on popup closed'] = function (assert, done) {
+    let defaultOpenWindowPref = getPref(OPEN_IN_NEW_WINDOW_PREF);
+    let defaultPopupPref = getPref(DISABLE_POPUP_PREF);
+
+    setPref(OPEN_IN_NEW_WINDOW_PREF, 2);
+    setPref(DISABLE_POPUP_PREF, false);
+
+    let tabID = "";
+
+    tabs.open({
+      url: 'about:blank',
+      onReady: function (tab) {
+        tabID = tab.id;
+        tab.attach({
+          contentScript: 'var popup = window.open("about:blank");' +
+                         'popup.close();'
+        });
+
+        setTimeout(function() {
```

Comment:
we are trying to avoid using `setTimeout`s in our tests when possible, because they lead to intermittent failures and slower tests.  We can avoid using `setTimeout` in this case by either using another method to open the window, or by listening to the [observer service for window notifications](https://developer.mozilla.org/en-US/docs/Observer_Notifications#Windows).